pkg/container/container.go

```diff
@@ -505,3 +514,56 @@ func (c *container) AddUnifiedResourcesFromAnnotations(annotationsMap map[string
 
 	return nil
 }
+
+// SpecSetProcessArgs sets the process args in the spec,
+// given the image information and passed-in container config
+func (c *container) SpecSetProcessArgs(imageOCIConfig *v1.Image) error {
+	// # Start the nginx container using the default command, but use custom
+	// arguments (arg1 .. argN) for that command.
+	// kubectl run nginx --image=nginx -- <arg1> <arg2> ... <argN>
+
+	// # Start the nginx container using a different command and custom arguments.
+	// kubectl run nginx --image=nginx --command -- <cmd> <arg1> ... <argN>
+
+	kubeCommands := c.config.Command
+	kubeArgs := c.config.Args
+
+	// merge image config and kube config
+	// same as docker does today...
+	if imageOCIConfig != nil {
+		if len(kubeCommands) == 0 {
+			if len(kubeArgs) == 0 {
+				kubeArgs = imageOCIConfig.Config.Cmd
+			}
+			if kubeCommands == nil {
+				kubeCommands = imageOCIConfig.Config.Entrypoint
+			}
+		}
+	}
+
+	if len(kubeCommands) == 0 && len(kubeArgs) == 0 {
+		return fmt.Errorf("no command specified")
+	}
+
+	// create entrypoint and args
+	var entrypoint string
+	var args []string
+	if len(kubeCommands) != 0 {
+		entrypoint = kubeCommands[0]
+		args = kubeCommands[1:]
+		args = append(args, kubeArgs...)
+	} else {
+		entrypoint = kubeArgs[0]
+		args = kubeArgs[1:]
+	}
```

Comment:
This is implicitely covered by the above statement:
```go
if len(kubeCommands) == 0 && len(kubeArgs) == 0 {
```
otherwise we could run into an index out of range if `len(kubeArgs) == 0 `.

How about combining both, for example:

```suggestion
	} else if len(kubeArgs) != 0 {
		entrypoint = kubeArgs[0]
		args = kubeArgs[1:]
	} else {
		return errors.New("no command specified")
	}
```

Reply:
good idea, updating